pkg/collector/corechecks/sbom/processor.go

```diff
@@ -179,14 +179,57 @@ func (p *processor) processContainerImagesRefresh(allImages []*workloadmeta.Cont
 	}
 }
 
-func (p *processor) processHostRefresh() {
-	if !p.hostSBOM {
+func (p *processor) processHostScanResult(result sbom.ScanResult) {
+	log.Debugf("processing host scanresult: %v", result)
+	sbom := &model.SBOMEntity{
+		Status:             model.SBOMStatus_SUCCESS,
+		Type:               model.SBOMSourceType_HOST_FILE_SYSTEM,
+		Id:                 p.hostname,
+		InUse:              true,
+		GeneratedAt:        timestamppb.New(result.CreatedAt),
+		GenerationDuration: convertDuration(result.Duration),
+	}
+
+	if result.Error != nil {
+		sbom.Sbom = &model.SBOMEntity_Error{
+			Error: result.Error.Error(),
+		}
+		sbom.Status = model.SBOMStatus_FAILED
+	} else {
+		log.Infof("Successfully generated SBOM for host: %v, %v", result.CreatedAt, result.Duration)
+
+		if p.hostCache != "" && p.hostCache == result.Report.ID() && result.CreatedAt.Sub(p.hostLastFullSBOM) < p.hostHeartbeatValidity {
+			sbom.Heartbeat = true
+		} else {
+			report, err := result.Report.ToCycloneDX()
+			if err != nil {
+				log.Errorf("Failed to extract SBOM from report: %s", err)
+				sbom.Sbom = &model.SBOMEntity_Error{
+					Error: err.Error(),
+				}
+				sbom.Status = model.SBOMStatus_FAILED
+			} else {
+				sbom.Sbom = &model.SBOMEntity_Cyclonedx{
+					Cyclonedx: convertBOM(report),
+				}
+			}
+
+			sbom.Hash = result.Report.ID()
+			p.hostCache = result.Report.ID()
+			p.hostLastFullSBOM = result.CreatedAt
+		}
+	}
+
+	p.queue <- sbom
+}
+
+func (p *processor) triggerHostScan(ch chan sbom.ScanResult) {
+	if !p.hostSBOM || ch == nil {
 		return
```

Comment:
IMO we need to split in 2 this `if` 
* `if !p.hostSBOM { return }` => because in that case it means this host scan is not enable
* `if ch == nil { // log error and return }` => because it is an unexpected case.

Also I suppose we can avoir calling this function if  `!p.hostSBOM == true` when we initialise the ticker

Reply:
I've split the `if` condition in 2. I am not sure what we can do to avoid calling this function and the potential cost in terms of readability

Reply:
I wanted to split it in two to only log a message in the second case which is a case that should not happen.
Indeed the code will be less readable but we will be able to differentiate the two cases.